snmp/tests/test_e2e_core_vs_python.py

```diff
@@ -126,6 +126,7 @@ def test_e2e_regex_match(dd_agent_check):
     assert_python_vs_core(dd_agent_check, config)
 
     config['init_config']['loader'] = 'core'
+    aggregator.reset()
```

Comment:
Really needed since you re-create an instance the line after?

Reply:
Yes, because `dd_agent_check(config, rate=True)` will append to the [global `aggregator`](https://github.com/DataDog/integrations-core/blob/2e5a4b3bd7d8ce02e7a84e46dfe76d084de08ebe/datadog_checks_dev/datadog_checks/dev/plugin/pytest.py#L34-L37) and also return the same `aggregator`.

Before this change, the aggregator contained each metric 4 times (2x from assert_python_vs_core, and 2x from dd_agent_check below. Note the 2x is due to `rate`).